documentation/src/main/asciidoc/user_guide/equivalence.adoc

```diff
@@ -1,4 +1,4 @@
-==  Storing objects (e.g. arrays) with custom Equivalence functions
+==  Object Equivalence
```

Comment:
I'm not sure this title works for people who don't already know what they're searching for. Maybe something like "Customizing equality comparison for keys and values" would be better?

IMO the next section is also not clear enough, as by stored "data" I understand the values, but not necessarily the keys. We should make it clear that a proper implementation of `equals()` and `hashCode()` is always required for keys, but it's only necessary for values if one intends to use conditional operations. It should also say more about how Infinispan uses `equals()` and `hashCode()` and what our requirements are, instead of how "users would like to call" one method or another...